packages/neos-ui/src/Containers/PrimaryToolbar/DimensionSwitcher/index.js

```diff
@@ -255,15 +257,58 @@ export default class DimensionSwitcher extends PureComponent {
         return null;
     }
 
+    getExistingDimensions() {
+        const allowed = this.props.allowedPresets
+        const currentDocumentNode = this.props.getNodeByContextPath(this.props.documentNode.contextPath)
+        const dimensionsWithVariants = currentDocumentNode?.otherNodeVariants;
+        if (!dimensionsWithVariants) {
+            return [currentDocumentNode.dimensions]
+        }
+
+        const existingDimensions = {};
+        Object.keys(allowed).forEach((dimensionName) => {
+            const dimensionValues = allowed[dimensionName];
+
+            existingDimensions[dimensionName] = [];
+            Array.from(dimensionValues).forEach((dimensionValue) => {
+                const result = [...dimensionsWithVariants, currentDocumentNode.dimensions].find((dimension) => {
+                    return dimension[dimensionName] === dimensionValue;
+                });
+                if (result) {
+                    existingDimensions[dimensionName].push(dimensionValue);
+                }
+            });
+        });
+
+        return existingDimensions;
+    }
+
     presetsForDimension(dimensionName) {
         const {contentDimensions, allowedPresets, i18nRegistry} = this.props;
         const dimensionConfiguration = $get(dimensionName, contentDimensions);
+        const existingDimensions = this.getExistingDimensions();
 
         return mapValues(dimensionConfiguration.presets,
             (presetConfiguration, presetName) => {
+                // if we do not know which dimensions exist, show all as existing
+                let existing = existingDimensions.length === 1 && existingDimensions[0] === undefined;
```

Comment:
Issue: Sometimes we do not get the existing dimensions (?switching between dimensions with overlapping defined values)
What should be displayed in this case.? (does not feel right, to mix showing and not showing the information)

(currently if we do not get the existing dimensions, all get shown as existing (white text))